packages/components/src/combobox-control/index.js

```diff
@@ -69,7 +69,7 @@ function ComboboxControl( {
 	} );
 
 	const currentOption = options.find( ( option ) => option.value === value );
-	const currentLabel = currentOption?.label ?? '';
+	const currentLabel = currentOption?.label ?? undefined;
```

Comment:
Changing the input's value from a string to `undefined` can sometimes introduce some unexpected consequences, as React can detect the component passing from uncontrolled to controlled mode.

This change may potentially need more investigation

Reply:
That is an easy fix. Happy to change today.